include/oneapi/dpl/internal/async_impl/glue_async_impl.h

```diff
@@ -0,0 +1,165 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_GLUE_ASYNC_IMPL_H
+#define _ONEDPL_GLUE_ASYNC_IMPL_H
+
+#if _ONEDPL_HETERO_BACKEND
+#    include "async_impl_hetero.h"
+#endif
+
+namespace oneapi
+{
+namespace dpl
+{
+namespace experimental
+{
+
+// [async.transform]
+template <class _ExecutionPolicy, class _ForwardIterator1, class _ForwardIterator2, class _UnaryOperation,
+          class... _Events>
+oneapi::dpl::__internal::__enable_if_device_execution_policy<
+    _ExecutionPolicy, oneapi::dpl::__internal::__future<_ForwardIterator2>, _Events...>
+transform_async(_ExecutionPolicy&& __exec, _ForwardIterator1 __first, _ForwardIterator1 __last,
+                _ForwardIterator2 __result, _UnaryOperation __op, _Events&&... __dependencies)
+{
+    wait_for_all(::std::forward<_Events>(__dependencies)...);
```

Comment:
Why we have a blocking call here?
In my vision (if I'm not mistaken) or we should just add dependency...